File: api-fchat/src/app/models/message.ts
import { DataTypes, Model } from "sequelize";
import sequelize from "../../config/sequelize";
import { UserInstance } from "./user";
import { ConversationInstance } from "./conversation";

interface MessageAttributes{
    id?: string;
    content?: string;
    date?: string;
    sender_id?: string;
    receiver_id?: string;
    conversation_id?: string
}

interface MessageInstance extends Model<MessageAttributes>, MessageAttributes{
    setSender: (user: UserInstance) => Promise<void>;
    setReceiver: (user: UserInstance) => Promise<void>;
    setConversation: (user: ConversationInstance) => Promise<void>;
}

const Message = sequelize.define<MessageInstance>('message', {
    id: {
        type: DataTypes.STRING,
        primaryKey: true
    },
    content: {
        type: DataTypes.STRING,
        allowNull: false,
        validate: {
            notEmpty: true
        }
    },
    date: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: sequelize.literal('CURRENT_TIMESTAMP'),
        validate: {
            notEmpty: true
        }
    },
},{
    createdAt: false,
    updatedAt: false
})

export default Message;

File: mobile_fchat/lib/state/models/message.dart

class Message{
  String? id;
  String? content;
  bool? is_read;
  String? date;
  String? sender_id;
  String? receiver_id;
  String? conversation_id;


  Message({
    this.id,
    this.content,
    this.is_read,
    this.date,
    this.sender_id,
    this.receiver_id,
    this.conversation_id,
  });

  Message.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    content = json['content'];
    is_read = json['is_read'];
    date = json['date'];
    sender_id = json['sender']['id'];
    receiver_id = json['receiver']['id'];
    conversation_id = json['conversation_id'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['content'] = content;
    data['is_read'] = is_read;
    data['date'] = date;
    data['sender_id'] = sender_id;
    data['receiver_id'] = receiver_id;
    data['conversation_id'] = conversation_id;
    return data;
  }
}